lib/samson/console_extensions.rb

```diff
@@ -0,0 +1,19 @@
+module Samson
+  module ConsoleExtensions
+    def login(role = Role::SUPER_ADMIN)
+      user = User.where(role_id: role.id).last!
```

Comment:
yeah true ... let's go with you have to give a user ... then you give
yourself and everything would be logged/executed correctly

On Wed, Jun 29, 2016 at 12:09 PM, Jon Moter <notifications@github.com>
wrote:

> In lib/samson/console_extensions.rb
> <https://github.com/zendesk/samson/pull/1113#discussion_r69008772>:
>
> > @@ -0,0 +1,19 @@
> > +module Samson
> > +  module ConsoleExtensions
> > +    def login(role = Role::SUPER_ADMIN)
> > +      user = User.where(role_id: role.id).last!
>
> Maybe if you just name it login_as_role then it would be more clear what
> you're doing.
>
> And add some docs on the purpose of the method. When I read it, I thought
> the idea was that if you did something like trigger a deployment, it would
> log the correct user as doing it. For compliance reasons it's important
> that we keep track of who's doing what, so I want to be careful about
> mechanism that allow impersonation.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1113/files/1f9f87078d669161b83dfba2f134fde079d9b8bb#r69008772>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAsZ0iYHaJQ8e3YOu1fedFqOaH2Cg0aks5qQsLQgaJpZM4JAwrS>
> .
>